### ISSUE TITLE
Update `@actions/core` to stop using `set-output` since it is deprecated

### ISSUE THREAD

**Issue by OrBin** [lots0logs/gh-action-get-changed-files#28]
This action [uses `@actions/core` version 1.2.6](https://github.com/lots0logs/gh-action-get-changed-files/blob/master/package.json#L10), which is using the `set-output` command, that is deprectated in GitHub Action [since October 11th](https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/).
`@actions/core` version 1.10.0 [fixes this issue](https://github.com/actions/toolkit/pull/1178).

Can you please upgrade this library to keep compatibility?

**Comment by MCKevmeister** [comment]
@lots0logs Any idea when this can be issue might be resolved? Otherwise, this action won't be able to be used beyond the [31 May 2023](https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/)

**Comment by MCKevmeister** [comment]
@lots0logs Please can we get an update? Is this an abandoned project?

**Comment by lots0logs** [comment]
@OrBin @MCKevmeister Please let me know if the latest release works okay for you. Thanks.

**Comment by MCKevmeister** [comment]
We have backlogged a ticket and will be address this before the end of the month. Thanks for releasing an update!

**Comment by OrBin** [comment]
Sorry @lots0logs, I do not use this action any more.

**Comment by Jeremy-NZ** [comment]
Hi @lots0logs,

it looks like you might need to upgrade some other related packages? Using the latest release I get the following error:

```
Prepare workflow directory
Prepare all required actions
Getting action download info
Download action repository 'actions/checkout@v3' (SHA:8e5e7e5ab8b370d6c3[29](https://github.com/genianz/genia-api/actions/runs/5020962694/jobs/9002909252#step:1:33)ec4802213[32](https://github.com/genianz/genia-api/actions/runs/5020962694/jobs/9002909252#step:1:36)ada57f0ab)
Download action repository 'actions/setup-node@v3' (SHA:64ed1c7eab4cce[33](https://github.com/genianz/genia-api/actions/runs/5020962694/jobs/9002909252#step:1:37)62f8c[34](https://github.com/genianz/genia-api/actions/runs/5020962694/jobs/9002909252#step:1:38)0dee64e5eaeef8f7c)
Download action repository 'lots0logs/gh-action-get-changed-files@2.2.0' (SHA:f11cd4567bcffc0bfd036526436e7475fa6d0418)
Error: System.ArgumentOutOfRangeException: Specified argument was out of the range of valid values. (Parameter ''using: node18' is not supported, use 'docker', 'node12' or 'node16' instead.')
   at GitHub.Runner.Worker.ActionManifestManager.ConvertRuns(IExecutionContext executionContext, TemplateContext templateContext, TemplateToken inputsToken, String fileRelativePath, MappingToken outputs)
   at GitHub.Runner.Worker.ActionManifestManager.Load(IExecutionContext executionContext, String manifestFile)
Error: Fail to load lots0logs/gh-action-get-changed-files/2.2.0/action.yml
```

**Comment by lots0logs** [comment]
Thanks. Fixed in 2.2.1 :heavy_check_mark:

**Comment by Jeremy-NZ** [comment]
Thanks @lots0logs. I just tested and that solves the node version param issue. Looks like there is another one though. I get this error now
```
Run lots0logs/gh-action-get-changed-files@2.2.1
/home/runner/work/_actions/lots0logs/gh-action-get-changed-files/2.2.1/dist/index.js:13100
	return gh.repos.getCommit(args);
	                ^

TypeError: Cannot read properties of undefined (reading 'getCommit')
    at fetchCommitData (/home/runner/work/_actions/lots0logs/gh-action-get-changed-files/2.2.1/dist/index.js:13100:18)
    at Array.map (<anonymous>)
    at /home/runner/work/_actions/lots0logs/gh-action-get-changed-files/2.2.1/dist/index.js:132[4](https://github.com/genianz/genia-api/actions/runs/5040035401/jobs/9038542331#step:4:5)1:22
    at processTicksAndRejections (node:internal/process/task_queues:96:[5](https://github.com/genianz/genia-api/actions/runs/5040035401/jobs/9038542331#step:4:6))
```

Same actions yaml works okay back on v2.1.4

**Comment by lots0logs** [comment]
@Jeremy-NZ Thanks for testing. I just released 2.2.2 which address that error.

**Comment by Jeremy-NZ** [comment]
That has done the trick. Thanks a million @lots0logs.
